rend3-routine/shaders/src/structures.wgsl

```diff
@@ -37,6 +37,40 @@ struct UniformData {
     resolution: vec2<u32>,
 }
 
+struct PerCameraUniformObjectData {
+    // TODO: use less space
+    model_view: mat4x4<f32>,
+    // TODO: use less space
+    model_view_proj: mat4x4<f32>,
+}
+
+struct PerCameraUniform {
+    // TODO: use less space
+    view: mat4x4<f32>,
+    // TODO: use less space
+    view_proj: mat4x4<f32>,
+    // The index of which shadow caster we are rendering for.
+    //
+    // This will be u32::MAX if we're rendering for a camera, not a shadow map.
+    shadow_index: u32,
+    frustum: Frustum,
+    resolution: vec2<f32>,
```

Comment:
We should ship the inverse half resolution as that's what we need in the shader.